services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_specs.py

```diff
@@ -41,32 +37,61 @@ def _inject_proxy_network_configuration(
     target_container_spec["networks"] = container_networks
 
 
+EnvKeyEqValueList = List[str]
+EnvVarsMap = Dict[str, Optional[str]]
+
+
+class _environment_section:
+    """the 'environment' field in a docker-compose can be either a dict (EnvVarsMap)
+    or a list of "key=value" (EnvKeyEqValueList)
+
+    These helpers can resolve parsing and exporting between these formats
+
+    SEE https://docs.docker.com/compose/compose-file/compose-file-v3/#environment
+    """
+
+    @staticmethod
+    def parse(environment: Union[EnvVarsMap, EnvKeyEqValueList]) -> EnvVarsMap:
+        envs = {}
+        if isinstance(environment, list):
+            for key_eq_value in environment:
+                assert isinstance(key_eq_value, str)  # nosec
+                key, value, *_ = key_eq_value.split("=") + [
```

Comment:
```suggestion
                key, value, *_ = key_eq_value.split("=", 1) + [
```

this will limit to only 1 element. You need to add a test with multiple equal signs for it to be sure

Reply:
thx i forgot that!